### ISSUE TITLE
Invalid mutation where value going throw and update all the table entries!

### ISSUE THREAD

**Issue by alaarihan** [hasura/graphql-engine#5701]
This mutation with this empty object shouldn't pass the graphql validation and shouldn't be executed at all but as you see it has executed and updated all the entries in the table with the value in the _inc
![API-Explorer-Hasura](https://user-images.githubusercontent.com/65740463/91774331-3d851900-ebf1-11ea-98e2-43f03b695888.png)

**Comment by alaarihan-temco** [comment]
And it's the same when I do _set, I thought it's only happens when I use _inc but it seems not limited to that.
This will update all the "page" entries status value to "ACTIVE" !
```
mutation MyMutation {
  update_page(where: {id: {}}, _set: {status: "ACTIVE"}) {
    affected_rows
  }
}
```

**Comment by alaarihan** [comment]
And it's the same when I do _set, I thought it's only happens when I use _inc but it seems not limited to that.
This will update all the "page" entries status value to "ACTIVE" !
```
mutation MyMutation {
  update_page(where: {id: {}}, _set: {status: "ACTIVE"}) {
    affected_rows
  }
}
```

**Comment by tirumaraiselvan** [comment]
Please see this issue: https://github.com/hasura/graphql-engine/issues/704

**Comment by alaarihan** [comment]
I don't agree with the solution that proposed there but thank you anyway..

I faced this problem because I assigned a promise as a value to "where.id" and I forgot to put await so the variable was a unresolved promise and it has been assumed as {} and the mutation has changed all the table entries instead of the intended id only.

**Comment by tirumaraiselvan** [comment]
@alaarihan Would be great if you can post your comment in #704 and close this issue so we can consolidate the discussion.
